src/main/kotlin/build/buf/intellij/BufPluginService.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.intellij
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.Service
+
+/**
+ * Application level service used as disposable.
+ * [Reference](https://plugins.jetbrains.com/docs/intellij/disposers.html?from=IncorrectParentDisposable#choosing-a-disposable-parent)
+ */
+@Service(Service.Level.APP)
+class BufPluginService : Disposable {
```

Comment:
Previously we were using the project as a parent disposable which is explicitly forbidden in the docs (and API checks). Update to create an application service used as a parent disposable when there isn't a better option available.